lib/watir/locators/element/locator.rb

```diff
@@ -141,57 +116,77 @@ def filter_elements(elements, selector, filter: :first)
           end
         end
 
-        def delete_filters_from(selector)
-          filter_selector = {}
+        def create_normalized_selector(filter)
+          return @normalized_selector if @normalized_selector
+          @driver_scope = ensure_scope_context
+
+          @normalized_selector = selector_builder.normalized_selector
+
+          process_label if @normalized_selector[:label]
+          return if @normalized_selector.nil?
```

Comment:
Would this only apply if `#process_label` did not find a label? To make it clear it's special logic for handling the labels, maybe we could do:

~~~~~~~~ruby
if @normalized_selector[:label]
  process_label 
  return if @normalized_selector.nil?
end
~~~~~~~~